packages/fx-core/src/common/local/localEnvManager.ts

```diff
@@ -37,38 +34,38 @@ export class LocalEnvManager {
     this.telemetry = telemetry;
   }
 
-  public async checkDependencies(projectSettings: ProjectSettings): Promise<
-    {
-      type: DepsType;
-      isInstalled: boolean;
-    }[]
-  > {
-    const depsLogger = new DepsLoggerAdapter(this.logger);
-    const depsTelemetry = new DepsTelemetryAdapter(this.telemetry);
-
-    const dependencies = this.getValidDeps(projectSettings);
-    const result = [];
-    for (const type of dependencies) {
-      const checker = CheckerFactory.createChecker(type, depsLogger, depsTelemetry);
-      const status = {
-        type: type,
-        isInstalled: await checker.isInstalled(),
-      };
-      result.push(status);
+  public getActiveDependencies(projectSettings: ProjectSettings): DepsType[] {
+    const depsTypes: DepsType[] = [];
+    const isSPFx = ProjectSettingsHelper.isSpfx(projectSettings);
+    const includeFrontend = ProjectSettingsHelper.includeFrontend(projectSettings);
+    const includeBackend = ProjectSettingsHelper.includeBackend(projectSettings);
+    const includeBot = ProjectSettingsHelper.includeBot(projectSettings);
+
+    // NodeJS
+    if (isSPFx) {
+      depsTypes.push(DepsType.SpfxNode);
+    } else if (includeBackend) {
+      depsTypes.push(DepsType.FunctionNode);
+    } else {
+      depsTypes.push(DepsType.AzureNode);
     }
-    return result;
-  }
 
-  public async checkAndResolveDependencies(
-    projectSettings: ProjectSettings
-  ): Promise<DependencyStatus[]> {
-    const depsLogger = new DepsLoggerAdapter(this.logger);
-    const depsTelemetry = new DepsTelemetryAdapter(this.telemetry);
-    const depsManager = new DepsManager(depsLogger, depsTelemetry);
+    // Dotnet
+    if (includeFrontend) {
+      depsTypes.push(DepsType.Dotnet);
+    }
 
-    return await depsManager.ensureDependencies(this.getValidDeps(projectSettings), {
-      fastFail: true,
-    });
+    // Function core tool
+    if (includeBackend) {
+      depsTypes.push(DepsType.FuncCoreTools);
+    }
+
+    // Ngrok
+    if (includeBot) {
+      depsTypes.push(DepsType.Ngrok);
```

Comment:
What if skipping ngrok?

Reply:
In the `Inner Loop New Workflow`,  
> localSettings.json 
> Remove frontend/trustDevCert (moved to vsc settings) 
> Remove bot/skipNgrok (moved to vsc settings) 

Can we remove it from here?

Reply:
The setting may be moved to another place but still configurable. So need to consider `skipNgrok` flag.

Reply:
Now all the vsc settings are loaded in vscodeChecker.
```
private async isEnabled(dep: DepsType): Promise<boolean> {
    switch (dep) {
      case DepsType.AzureNode:
      case DepsType.SpfxNode:
        return checkerEnabled(VSCodeDepsChecker.validateNodeVersionKey);
      case DepsType.FunctionNode:
        return checkerEnabled(VSCodeDepsChecker.validateNodeVersionKey) && (await hasFunction());
      case DepsType.Dotnet:
        return checkerEnabled(VSCodeDepsChecker.validateDotnetSdkKey);
      case DepsType.FuncCoreTools:
        return checkerEnabled(VSCodeDepsChecker.validateFuncCoreToolsKey) && (await hasFunction());
      case DepsType.Ngrok:
        // TODO: remove hasBot and get hasNgrok from workspace configuration
        return (await hasBot()) && (await hasNgrok());
      default:
        return false;
    }
```
Can we only generate the deps types according to projectSettings in localEnvManager?

Reply:
I see, so this is to generate the deps list and still need `vscodeChecker`/`cliChecker` to double check all flags.